grub-efi/docker-create-grub-efi-binaries

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+
+set -e
+
+GRUB_VERSION=${GRUB_VERSION:-2.04}
+
+sudo rm -rf output
+
+for arch in x86-64 arm aarch64; do
+    mkdir -p output/$arch
+    echo -n "Building binaries for $arch..."
+    docker_build_ret=0
+    docker build \
+           -f Dockerfile.$arch \
+           -t grub-efi:$arch \
+           --build-arg GRUB_VERSION=${GRUB_VERSION} \
+           . > /tmp/docker-build.log || docker_build_ret=$?
+    if [ $docker_build_ret -ne 0 ]; then
+        echo " failed!"
+        tail -n 100 /tmp/docker-build.log
+        exit $docker_build_ret
+    else
+        echo " done"
+    fi
+    echo -n "Extracting binaries for $arch..."
+    docker run --rm --entrypoint "/bin/sh" \
+        -v $(pwd)/output/$arch:/export grub-efi:$arch \
+        -c "cp -r /install/* /export"
+    echo " done"
+done
+
+sudo chown -R $(id -u):$(id -g) output/*
+
+echo "Generating EFI boot images"
+
+output_dir=$(pwd)/output
+grub_mkimage_native=${output_dir}/x86-64/bin/grub-mkimage
+
+set -x
+
+$grub_mkimage_native \
+    -p /EFI/BOOT \
+    -d ${output_dir}/x86-64/lib/grub/x86_64-efi \
+    -O x86_64-efi \
+    -o ${output_dir}/grub-efi-bootx64.efi
+
+$grub_mkimage_native \
+    -p /EFI/BOOT \
+    -d ${output_dir}/arm/lib/grub/arm-efi \
+    -O arm-efi \
+    -o ${output_dir}/grub-efi-bootarm.efi
```

Comment:
So you have not tried running it anywhere? I suspect you need to give it some module names here, or you will get an "empty" bootloader, capable of essentially nothing. Currently ours are listed [here](https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-bsp/grub/grub-mender.inc#L4), but this isn't a very good place, and ideally this list should be moved to grub-mender-grubenv instead.

Reply:
I didn't know about these modules. I'll investigate a bit more.

But the fact is that it works, at least on BBB. I did a full rootfs update, and the device did reboot into the new partition and all. There were some errors printed to the console so I will try the same experiment with a Debian 9.5 based images to confirm.

Reply:
> But the fact is that it works, at least on BBB. I did a full rootfs update, and the device did reboot into the new partition and all. [...]

That is incorrect: the board boots into the old partition. That was an oversight as I have been experimenting with standalone install only.

More investigation is required. Thanks for pointing out that this was suspicious.

Reply:
I added the missing modules as you pointed out, see force-pushed code.

At least for BBB and Debian 9.5 it works now :)